### PR TITLE
Add support for S3 in spark-hdfs

### PR DESCRIPTION
**Pull request by rabbitonweb**
Fixes https://github.com/quasar-analytics/quasar/issues/2264

### CODE REVIEW

File: sparkcore/src/main/scala/quasar/physical/sparkcore/fs/hdfs/SparkHdfs.scala [hunk 1]
```diff
@@ -184,19 +192,32 @@ object SparkHdfs extends SparkCore with ChrootedInterpreter {
             .fold(liftErr("'rootPath' is not a valid path").left[ADir])(_.right[DefinitionError]).point[Task])
         )
 
+    val s3InfoOrErr: DefErrT[Task, Option[S3Info]] =
+      uriOrErr.map(uri => for {
```
Comment (drostron) [inline sparkcore/src/main/scala/quasar/physical/sparkcore/fs/hdfs/SparkHdfs.scala]: ⛳️
```scala
val s3InfoOrErr: DefErrT[Task, Option[S3Info]] =
  uriOrErr.map(uri =>
    (uri.params.get("s3AccessKey") ⊛ uri.params.get("s3SecretKey") ⊛ uri.params.get("s3Endpoint"))(S3Info))
```

File: sparkcore/src/main/scala/quasar/physical/sparkcore/fs/hdfs/SparkHdfs.scala [hunk 2]
```diff
@@ -184,19 +192,32 @@ object SparkHdfs extends SparkCore with ChrootedInterpreter {
...
+        accessKey <- uri.params.get("s3AccessKey")
+        secretKey <- uri.params.get("s3SecretKey")
+        endpoint  <- uri.params.get("s3Endpoint")
+      } yield S3Info(accessKey, secretKey, endpoint))
+
     for {
```
Comment (drostron) [inline sparkcore/src/main/scala/quasar/physical/sparkcore/fs/hdfs/SparkHdfs.scala]: Could accumulate errors
```scala
(sparkConfOrErr ⊛ hdfsUrlOrErr ⊛ rootPathOrErr ⊛ s3InfoOrErr)(HdfsConfig)
```

**Comment by drostron** [comment]
Needs a semver label

**Comment by rabbitonweb** [comment]
@drostron now, question, should I consider this a new feature? I guess so...

**Comment by drostron** [comment]
@rabbitonweb I'm amenable to `version: feature`

**Comment by rabbitonweb** [comment]
@drostron done

**Pull request merged**